### PR TITLE
Merge `develop` into `master`

### PR DESCRIPTION
**Pull request by donn**
I rebased the develop branch so it can be merged to master.

The following significant feature updates will be merged into master:

Ahmad Nofal <ahmednofal@aucegypt.edu>
\+ Ability to run parts of the flow, with the ability to resume.

Manar Abdelaty <manarabdelatty@aucegypt.edu>
\+ Make IO min distance an environment variable
\~ Various Documentation Updates

Matthew Ballance <matt.ballance@gmail.com>
\+ Support podman (RHEL8) as a Docker replacement
\+ Add include paths when processing top-level and blackbox verilog files

Matt Venn <matt@mattvenn.net>
\+ Dot file to examine the techmap process
    * (Done before optimization so results are inspectable for small designs.)

Mohamed Gaber <me@donn.website>
\+ Ability to run OpenLane without Docker on Ubuntu 20.04
\+ Added ability to package issues with OpenROAD scripts as a small reproducible
\+ Dockerfiles contain Repo and Commit 
\+ Fixed installed Klayout versions, updated disclaimer
\+ Update libtrim.pl to work with reasonably modern versions of perl
\~ Fixed make fastest_test_set
\~ Various Documentation Updates
\~ Screenshots Turned Off By Default
    * They crash in some environments.

Mohamed Gaber <me@donn.website>
Osama Hammad <osama21@aucegypt.edu>
\+ Migrate to a more current version of OpenROAD (#433) (**!!**)
\- Remove all standalone utilities that were integrated into OpenROAD

# Known Issues
None.

### CODE REVIEW

**Comment by donn** [comment]
As this is a huge, huge delta, I request that it is not merge without approval from **all reviewers**.

File: docker_build/docker/openphysyn/Dockerfile
Comment (maliberty) [inline docker_build/docker/openphysyn/Dockerfile:71]: Is openphysyn still part of OL?  I thought you were moving away from it.

File: scripts/openroad/or_cts.tcl
Comment (maliberty) [inline scripts/openroad/or_cts.tcl:46]: Do not called functions outside the global namespace.  I don't think these commands are required but if they are you should use configure_cts_characterization.  We do not use them in ORFS.

File: scripts/openroad/or_droute.tcl
Comment (maliberty) [inline scripts/openroad/or_droute.tcl:27]: All the options in -param are now available as command options so there is no need to use the param file anymore.

File: scripts/openroad/or_groute.tcl
Comment (maliberty) [inline scripts/openroad/or_groute.tcl:63]: The command is global_route.  Please remove all usage of grt::*

File: scripts/openroad/or_replace.tcl
Comment (maliberty) [inline scripts/openroad/or_replace.tcl:49]: Do not use gpl::*, use the proper commands.

**Review by maliberty — APPROVED** [review]
I don't feel my comments should stop this PR from being merged but they do need to be addressed going forward.

**Comment by donn** [comment]
@maliberty Yeah, all your review points are valid. To give context on the whole internal commands thing, back when we were using the tools standalone, the internal commands were global namespace. We kept this initial phase of the transition as 1:1 as humanly possible.

File: scripts/tcl_commands/all.tcl
Comment (Manarabdelaty) [inline scripts/tcl_commands/all.tcl:407]: I think `scripts/new_tracks.py` need to be replaced with `$::env(SCRIPTS_DIR)/new_tracks.py` because not all designs are run from openlane's top level directory

**Comment by donn** [comment]
Well since four people apparently have something better to do. :)

**Review by Manarabdelaty — COMMENTED** [review]
@kareefardi was having an issue with the ioplace.py script. Looks like the opendb apis were update, the ioplace.py script is failing on 
```
 File "/openLANE_flow/scripts/io_place.py", line 288, in <module>
    origin, count, step = odb.new_int(0), odb.new_int(0), odb.new_int(0)
AttributeError: module 'opendb' has no attribute 'new_int'
```

File: configuration/README.md
Comment (osamahammad21) [inline configuration/README.md:119]: I think we should output a warning if a deprecated configuration option is used. This will warn users instead of trying to debug why the option isn't effective.

File: configuration/README.md
Comment (osamahammad21) [inline configuration/README.md:75]: deprecation warning

File: configuration/README.md
Comment (osamahammad21) [inline configuration/README.md:158]: ditto

**Comment by donn** [comment]
> @kareefardi was having an issue with the ioplace.py script. Looks like the opendb apis were update, the ioplace.py script is failing on

Is that not part of our CI?

File: scripts/tcl_commands/all.tcl
Comment (donn) [reply scripts/tcl_commands/all.tcl:407]: Will tackle.

File: configuration/floorplan.tcl
Comment (osamahammad21) [inline configuration/floorplan.tcl:52]: I don't think we should keep it as an option since disabling it will crash the run. If there is no current usage, please remove and update OpenPDK's pdn configuration

File: designs/inverter/config.tcl
Comment (osamahammad21) [inline designs/inverter/config.tcl:13]: can you clarify why?

File: configuration/floorplan.tcl
Comment (osamahammad21) [inline configuration/floorplan.tcl:43]: please remove. This option is deprecated

File: scripts/openroad/or_groute.tcl
Comment (osamahammad21) [inline scripts/openroad/or_groute.tcl:57]: please remove

**Comment by Manarabdelaty** [comment]
> > @kareefardi was having an issue with the ioplace.py script. Looks like the opendb apis were update, the ioplace.py script is failing on
> 
> Is that not part of our CI?

No, I don't think the CI is testing the custom io placement. If you uncomment the `FP_PIN_ORDER_CFG`  variable here https://github.com/The-OpenROAD-Project/OpenLane/blob/master/designs/spm/config.tcl#L11 and run spm you should be able to observe the failure

**Review by osamahammad21 — COMMENTED** [review]
These are some small comments in addition to Matt's review. I think the most important think is to use OpenROAD's exposed command instead of submodules' commands.

**Pull request merged**
